src/Sarif/NotYetGenerated/SarifRewritingVisitor.cs

```diff
@@ -125,14 +125,56 @@ public virtual object VisitActual(ISarifNode node)
             }
         }
 
+
         private T VisitNullChecked<T>(T node) where T : class, ISarifNode
+        {
+            string emptyKey = null;
+            return VisitNullChecked<T>(node, ref emptyKey);
+        }
+
+
+            private T VisitNullChecked<T>(T node, ref string key) where T : class, ISarifNode
         {
             if (node == null)
             {
                 return null;
             }
 
-            return (T)Visit(node);
+            if (key == null)
+            {
+                return (T)Visit(node);
+            }
+
+            return (T)VisitDictionaryEntry(node, ref key);
+        }
+
+        private T VisitDictionaryEntry(ISarifNode node, ref string key) where T : class, ISarifNode
+        {
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            switch (node.SarifNodeKind)
+            {
+                case SarifNodeKind.FileData:
```

Comment:
>FileData [](start = 35, length = 8)

I guess the generator could identify the set of types that can occur as dictionary entries, and generate a `case` for each, but I'm going to consider whether there's a more generic way of doing this.